kod/object/active/holder/nomoveon/battler/player/user.kod

```diff
@@ -2418,8 +2419,39 @@ messages:
       rarity = 0;
       if IsClass(what,&Item)
       {
-         rarity = Send(what,@GetRarity);
+         if bShow_All  % Used for identify and reveal "look" rarity assignment
```

Comment:
Yikes, this isn't what I expected.  This is equivalent to the previous version?  I thought you could put a single check here for bShow_All, not reproducing the entire list of options (which we clearly don't want to do).

Can you explain the intent here?

Reply:
Sorry.  My intent was to skip the "unidentified" section of the rarity grade check.

The reason is that if I cast identify on an unidentified item, it will go through the GetRarity function and return "unidentified" because ... it is.

So what I want to do is skip that check, to evaluate everything else except unidentified rarity.  With the change I just put in, if bShow_all is true, it does all the checks except unidentified.

I don't really know how to fix this without modifying GetRarity to accept the parameter of bShow_all or create a second function like this where the unidentified check isn't done.

I backed myself into a corner without knowing it, because if we were doing flags instead of int, then I could pass the full rarity value with (unidentified + cursed or + itematts) and then just parse it on the client.

I hope I'm explaining my intent correctly - it is complicated.  But in short, I want to check for everything except IsIdentified if bShow_All is TRUE.

Reply:
I see.. I guess I didn't understand this before.  The previous version was much better than this level of code duplication.  But perhaps you can come up with a better name for the parameter than the meaningless "bShow_All", and also add documentation for it.